core/templates/dev/head/services/UserService.ts

```diff
@@ -17,48 +17,56 @@
  */
 
 oppia.factory('UserService', [
-  '$http', '$q', '$window', 'UrlInterpolationService', 'UserInfoObjectFactory',
-  'DEFAULT_PROFILE_IMAGE_PATH',
-  function($http, $q, $window, UrlInterpolationService, UserInfoObjectFactory,
-      DEFAULT_PROFILE_IMAGE_PATH) {
+  '$http', '$q', '$window', 'UrlInterpolationService',
+  'UserInfoObjectFactory', 'DEFAULT_PROFILE_IMAGE_PATH',
+  function($http, $q, $window, UrlInterpolationService,
+      UserInfoObjectFactory, DEFAULT_PROFILE_IMAGE_PATH) {
     var PREFERENCES_DATA_URL = '/preferenceshandler/data';
 
     var userInfo = null;
 
     var getUserInfoAsync = function() {
-      if (GLOBALS.userIsLoggedIn) {
-        if (userInfo) {
-          return $q.resolve(userInfo);
-        }
-        return $http.get(
-          '/userinfohandler'
-        ).then(function(response) {
-          userInfo = UserInfoObjectFactory.createFromBackendDict(response.data);
-          return userInfo;
-        });
-      } else {
-        return $q.resolve(UserInfoObjectFactory.createDefault());
+      var currentUrl = $window.location.href;
+      if (currentUrl.includes('signup')) {
+        userInfo = UserInfoObjectFactory.createDefault();
+        return $q.resolve(userInfo);
+      }
+      if (userInfo) {
+        return $q.resolve(userInfo);
       }
+      return $http.get(
+        '/userinfohandler'
+      ).then(function(response) {
+        if (response.data.user_is_logged_in) {
+          userInfo = (
+            UserInfoObjectFactory.createFromBackendDict(response.data));
+        } else {
+          userInfo = UserInfoObjectFactory.createDefault();
+        }
+        return $q.resolve(userInfo);
+      });
     };
 
     return {
       getProfileImageDataUrlAsync: function() {
         var profilePictureDataUrl = (
           UrlInterpolationService.getStaticImageUrl(
             DEFAULT_PROFILE_IMAGE_PATH));
-
-        if (GLOBALS.userIsLoggedIn) {
-          return $http.get(
-            '/preferenceshandler/profile_picture'
-          ).then(function(response) {
-            if (response.data.profile_picture_data_url) {
-              profilePictureDataUrl = response.data.profile_picture_data_url;
-            }
-            return profilePictureDataUrl;
-          });
-        } else {
-          return $q.resolve(profilePictureDataUrl);
-        }
+        return getUserInfoAsync().then(function(userInfo) {
+          if (userInfo.isLoggedIn()) {
+            return $http.get(
+              '/preferenceshandler/profile_picture'
+            ).then(function(response) {
+              if (response.data.profile_picture_data_url) {
```

Comment:
Is there any issue with asking this data in the userInfo call? 
/cc @vojtechjelinek